x/ibc/23-commitment/merkle.go

```diff
@@ -0,0 +1,145 @@
+package commitment
+
+import (
+	"strings"
+
+	"github.com/tendermint/tendermint/crypto/merkle"
+
+	"github.com/cosmos/cosmos-sdk/store/rootmulti"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+)
+
+// ICS 023 Merkle Types Implementation
+//
+// This file defines Merkle commitment types that implements ICS 023.
+
+// Merkle proof implementation of the Proof interface
+// Applied on SDK-based IBC implementation
+var _ RootI = Root{}
+
+// Root defines a merkle root hash.
+// In the Cosmos SDK, the AppHash of a block header becomes the Root.
+type Root struct {
+	Hash []byte `json:"hash" yaml:"hash"`
+}
+
+// NewRoot constructs a new Root
+func NewRoot(hash []byte) Root {
+	return Root{
+		Hash: hash,
+	}
+}
+
+// GetCommitmentType implements RootI interface
+func (Root) GetCommitmentType() Type {
+	return Merkle
+}
+
+// GetHash implements RootI interface
+func (r Root) GetHash() []byte {
+	return r.Hash
+}
+
+var _ PrefixI = Prefix{}
+
+// Prefix is merkle path prefixed to the key.
+// The constructed key from the Path and the key will be append(Path.KeyPath, append(Path.KeyPrefix, key...))
+type Prefix struct {
+	KeyPrefix []byte `json:"key_prefix" yaml:"key_prefix"` // byte slice prefixed before the key
+}
+
+// NewPrefix constructs new Prefix instance
+func NewPrefix(keyPrefix []byte) Prefix {
+	return Prefix{
+		KeyPrefix: keyPrefix,
+	}
+}
+
+// GetCommitmentType implements PrefixI
+func (Prefix) GetCommitmentType() Type {
+	return Merkle
+}
+
+// Bytes returns the key prefix bytes
+func (p Prefix) Bytes() []byte {
+	return p.KeyPrefix
+}
+
+var _ PathI = Path{}
+
+// Path is the path used to verify commitment proofs, which can be an arbitrary
+// structured object (defined by a commitment type).
+type Path struct {
+	KeyPath merkle.KeyPath `json:"key_path" yaml:"key_path"` // byte slice prefixed before the key
+}
+
+// NewPath creates a new CommitmentPath instance
+func NewPath(keyPathStr []string) Path {
+	merkleKeyPath := merkle.KeyPath{}
+	for _, keyStr := range keyPathStr {
+		merkleKeyPath = merkleKeyPath.AppendKey([]byte(keyStr), merkle.KeyEncodingHex)
```

Comment:
Why is each path element being encoded with hex? It's not clear to me looking at the rootmulti store that this is how it is being done?

Should it not be `merkle.KeyEncodingURL`